src/NHibernate.Test/DynamicEntity/DataProxyHandler.cs

```diff
@@ -53,6 +53,15 @@ public object Intercept(InvocationInfo info)
 			{
 				return GetHashCode();
 			}
+			else if ("Equals".Equals(methodName))
```

Comment:
In other words, you suggest to simplify this test fix by just calling `RuntimeHelpers.Equals(this, obj)` instead of bothering calling a base implementation of a class we are not directly deriving from?

Ok.